tensorrt_llm/_torch/attention_backend/trtllm.py

```
@@ -1103,48 +1104,68 @@ def update_spec_dec_param(
+            # Prepare the spec-dec mask, position offset and generation length for static tree of dynamic tree.
+            # We only prepare the spec-dec mask, position offset and generation length for the target model here.
+            # For the drafter model, we will prepare them in the drafting loops.
+            is_target_model = not spec_metadata.is_draft_model
+            is_using_tree = self.is_spec_dec_tree or self.is_spec_dec_dynamic_tree
+            if is_target_model and is_using_tree:
+                assert spec_metadata.spec_dec_mode.is_eagle3(
+                ), "Tree decoding is only supported for Eagle3 now"
+                # If is the dynamic tree
+                if self.is_spec_dec_dynamic_tree:
+                    # TODO: add dynamic tree logic
+                    assert False, "Dynamic tree is not supported yet"
```

Ruff (0.14.1)

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)